vms/platformvm/txs/fee/calculator_api.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fee
+
+import (
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+// Calculator is a wrapper to exposed a more user friendly API than txs.Visitor allows
+// backend is not embedded to avoid exposing unnecessary methods
+type Calculator struct {
+	b backend
+}
+
+func (c *Calculator) CalculateFee(tx txs.UnsignedTx, creds []verify.Verifiable) (uint64, error) {
```

Comment:
this is one of the main changes of this PR.
`CalculateFee` returns an error, which will be needed as soon as dynamic fees are introduced.
Moreover `chainTime` parameter has been moved to the static fee calculator constructor. There are two reasons for this:

- Dynamic fees calcualtor won't need the chain time as a `CalculateFee` parameter; a more complex `commonfee.Calculator` object will be introduced, which tracks gas prices and cumulated consumed gas for both P and X chain. 
- Moreover the calculator will be instantiated one per block (currently once per each transaction in a block), and every block transactions will have the same chainTime (the block timestamp), hence we can avoid passing it explicitly at each CalculateFee invocation.

Reply:
I think the unsigned + creds split is a left over from the visitor pattern. Dropped in favour of the signed tx parameter.
fees will be a function of the whole tx, not just signers. For instance tx size, referenced utxos counts will be turned into gas, then. fees